File: KanbanBoardApi/Models/Users/UserModels.cs
namespace KanbanBoardApi.Models.Users;

public record UserModel
{
    public int Id { get; init; }
    public required string Email { get; init; }
    public required string FirstName { get; init; }
    public required string LastName { get; init; }
    public required string PhotoUrl { get; init; }

    public string GetName() => $"{FirstName} {LastName}";
    public string GetNameWithEmail() => $"{FirstName} {LastName} ({Email})";
}

File: kanban-board-app/src/models/userModels.ts
interface User {
  id: number;
  email: string;
  firstName: string;
  lastName: string;
  photoUrl: string;
}

const getFullName = (user?: User | null) => (user ? `${user.firstName} ${user.lastName}`.trim() : '');

export { type User, getFullName };
